src/devices/Card/Ultralight/UltralightCard.cs

```diff
@@ -0,0 +1,992 @@
+﻿using System;
+using Iot.Device.Common;
+using Iot.Device.Ndef;
+using Microsoft.Extensions.Logging;
+
+namespace Iot.Device.Card.Ultralight
+{
+    /// <summary>
+    /// A Ultralight card class
+    /// </summary>
+    public class UltralightCard
+    {
+        /// <summary>
+        /// Default password used for write and read
+        /// </summary>
+        public static readonly byte[] DefaultPassword = new byte[4] { 0xFF, 0xFF, 0xFF, 0xFF };
+
+        // This is the actual RFID reader
+        private readonly CardTransceiver _rfid;
+        private readonly ILogger _logger;
+        private byte[]? _pack;
+
+        private byte _endAddress;
+
+        /// <summary>
+        /// The tag number detected by the reader, only 1 or 2
+        /// </summary>
+        public byte Target { get; set; }
+
+        /// <summary>
+        /// The NDEF capacity in bytes
+        /// </summary>
+        public int NdefCapacity { get; internal set; }
+
+        /// <summary>
+        /// The type of card
+        /// </summary>
+        public UltralightCardType UltralightCardType { get; internal set; }
+
+        /// <summary>
+        /// The block number to authenticate or read or write
+        /// </summary>
+        public byte BlockNumber { get; set; }
+
+        /// <summary>
+        /// The Data which has been read or to write for the specific block
+        /// </summary>
+        public byte[]? Data { get; set; }
+
+        /// <summary>
+        /// The command to execute on the card
+        /// </summary>
+        public UltralightCommand Command { get; set; }
+
+        /// <summary>
+        /// The counter to read or increment
+        /// </summary>
+        public byte Counter { get; set; }
+
+        /// <summary>
+        /// Authentication key
+        /// </summary>
+        public byte[] AuthenticationKey { get; set; } = DefaultPassword;
+
+        /// <summary>
+        /// UUID is the Serial Number, called MAC sometimes
+        /// </summary>
+        public byte[]? SerialNumber { get; set; }
```

Comment:
This is writable?

Reply:
Yes if you want to create a virtual card.

Reply:
consider making all `byte[]` properties non-nullable and internally return/set empty buffer on null